edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.IoTHub/EdgeAgentConnection.cs

```diff
@@ -119,6 +119,28 @@ public async Task UpdateReportedPropertiesAsync(TwinCollection patch)
             }
         }
 
+        public async Task SendEventAsync(Message message)
+        {
+            Events.UpdatingReportedProperties();
+            try
+            {
+                Option<IModuleClient> moduleClient = this.moduleConnection.GetModuleClient();
+                if (!moduleClient.HasValue)
+                {
+                    Events.SendEventClientEmpty();
+                    return;
+                }
```

Comment:
I see you followed the pattern here. Is this protection against the moduleClient being disposed?

Reply:
I don't think this should ever be empty, thats why it errors. I'm not sure, but I just followed the pattern.